OOps/bus.c

```diff
@@ -2164,3 +2274,419 @@ int32_t outvalsetSgo(CSOUND *csound, OUTVAL *p)
     // if (ans==OK) ans = koutval(csound,p);
     return ans;
 }
+
+/* ARRAY channels  implementation - VL 7.08.24 */
+static void copy_array(ARRAYDAT *out, const ARRAYDAT *in, spin_lock_t *lock) {
```

Comment:
Pretty sure this won't work for all array types; see array_copy_value in csound_standard_types.c. We should reuse the type system's copy mechanism here, otherwise I think we'll get unexpected data sharing.

Reply:
absolutely
ac236c38b